src/BlazorBoilerplate.Server/Startup.cs

```diff
@@ -49,8 +49,14 @@ public void ConfigureServices(IServiceCollection services)
         {
             string migrationsAssembly = typeof(Startup).GetTypeInfo().Assembly.GetName().Name;
 
+            //assemble the connection string from docker-compose variables
+            string constring = $"Server={Configuration.GetValue<string>("DOCKER_COMPOSE_SQL")};Database=master;User={Configuration.GetValue<string>("MSSQL_USER")};Password={Configuration.GetValue<string>("SA_PASSWORD")}";
             services.AddDbContext<ApplicationDbContext>(options => {
-                if (Convert.ToBoolean(Configuration["BlazorBoilerplate:UseSqlServer"] ?? "false"))
+                if (!String.IsNullOrWhiteSpace(Configuration.GetValue<string>("DOCKER_COMPOSE_SQL")))
+                {
+                    options.UseSqlServer(constring);  // SQL Server from docker-compose
+                }
+                else if (Convert.ToBoolean(Configuration["BlazorBoilerplate:UseSqlServer"] ?? "false"))
                 {
                     options.UseSqlServer(Configuration.GetConnectionString("DefaultConnection"), sql => sql.MigrationsAssembly(migrationsAssembly));//SQL Server Database
                 }
```

Comment:
It might be better idea to store the SqlServerConnectionString as whole in appsettings.json, and than set the same environment variable in docker-compose.
During startup, first value from appsettings.json is read, next if the save key is present in environment variables set in docker-compose, the value is overriden.

Storing the whole connectionstring (not only the server, user, password) gives freehand in configuring it for example using windows trusted auth.

Having the code itself know anything about the docker ("DOCKER_COMPOSE_SQL") is a bad idea, the code should be agnostic on the run environment

Reply:
Not sure if I did that right but I "resolved the conversation" I will merge this PR and if the suggested changes did not come over then we can do another PR for @acid12  suggestions. Thanks to both of your for your contributions!